lib/RequestRouter.js

```diff
@@ -496,8 +496,14 @@ RequestRouter.prototype.addPropertyError = function(property, errorCode) {
         }
 
     }
+
+    //Remove the index of the array if it's added to the end of the property name
+    //so we can find the parent property in the schema
+    //Ex: paramArray.0 will become paramArray
+    var propertyTemp = _.clone(property).replace(/(\.[0-9]+)$/, '');
```

Comment:
If possible, I suggest sticking with the monocle-props syntax. Check out the monocle-props module for useful utility methods like `.set()` and `.get()`